### PR TITLE
GPG sign

### PR DESCRIPTION
**Pull request by Stayer**
### Description of the Change
This patch allows to sign all artifact files using gnupg utility

### Benefits
Users can trust and check to all published binaries. Public PGP key are published on pgp.mit.edu

### Possible Drawbacks 
None

### CODE REVIEW

File: .jenkinsci/artifacts.groovy [hunk 1]
```diff
@@ -11,17 +11,26 @@ def uploadArtifacts(filePaths, uploadPath, artifactServers=['artifact.soramitsu.
...
-  filePathsConverted.each {
-    sh "echo put ${it} $uploadPath >> \$(pwd)/batch.txt;"
-    sh "$shaSumBinary ${it} | cut -d' ' -f1 > \$(pwd)/\$(basename ${it}).sha256"
-    sh "$md5SumBinary ${it} | cut -d' ' -f1 > \$(pwd)/\$(basename ${it}).md5"
-    sh "echo put \$(pwd)/\$(basename ${it}).sha256 $uploadPath >> \$(pwd)/batch.txt;"
-    sh "echo put \$(pwd)/\$(basename ${it}).md5 $uploadPath >> \$(pwd)/batch.txt;"
+
+  withCredentials([file(credentialsId: 'ci_gpg_privkey', variable: 'CI_GPG_PRIVKEY'), string(credentialsId: 'ci_gpg_masterkey', variable: 'CI_GPG_MASTERKEY')]) {
+
+    sh "gpg --allow-secret-key-import --import ${CI_GPG_PRIVKEY}"
```
Comment (bakhtin) [inline .jenkinsci/artifacts.groovy]: `--allow-secret-key-import` is an obsolete as stated in `man`. Should we really use it?

File: docker/develop/Dockerfile [hunk 2]
```diff
@@ -263,6 +263,11 @@ RUN set -e; \
 RUN set -e; \
     curl -L -o /tmp/lcov-1.13.tar.gz https://github.com/linux-test-project/lcov/releases/download/v1.13/lcov-1.13.tar.gz; cd /tmp; tar zxf lcov-1.13.tar.gz; cd lcov-1.13; make install
 
+# install gpg
```
Comment (bakhtin) [inline docker/develop/Dockerfile]: Please move `gnupg` installation to the top section of the file

File: .jenkinsci/artifacts.groovy [hunk 3]
```diff
@@ -11,17 +11,26 @@ def uploadArtifacts(filePaths, uploadPath, artifactServers=['artifact.soramitsu.
   }
   def shaSumBinary = 'sha256sum'
   def md5SumBinary = 'md5sum'
+  def gpgKeyBinary = 'gpg --armor --detach-sign --default-key 9C49B055 --batch --yes --passphrase-fd 0'
```
Comment (bakhtin) [inline .jenkinsci/artifacts.groovy]: Is it possible to dynamically parse key name instead of hardcoding it?

**Comment by bakhtin** [comment]
Consider rebasing commits that were not signed-off. You'd better squash most of your commits either as they have the same name.

File: .jenkinsci/artifacts.groovy [hunk 4]
```diff
@@ -11,17 +11,26 @@ def uploadArtifacts(filePaths, uploadPath, artifactServers=['artifact.soramitsu.
...
-  filePathsConverted.each {
-    sh "echo put ${it} $uploadPath >> \$(pwd)/batch.txt;"
-    sh "$shaSumBinary ${it} | cut -d' ' -f1 > \$(pwd)/\$(basename ${it}).sha256"
-    sh "$md5SumBinary ${it} | cut -d' ' -f1 > \$(pwd)/\$(basename ${it}).md5"
-    sh "echo put \$(pwd)/\$(basename ${it}).sha256 $uploadPath >> \$(pwd)/batch.txt;"
-    sh "echo put \$(pwd)/\$(basename ${it}).md5 $uploadPath >> \$(pwd)/batch.txt;"
+
+  withCredentials([file(credentialsId: 'ci_gpg_privkey', variable: 'CI_GPG_PRIVKEY'), string(credentialsId: 'ci_gpg_masterkey', variable: 'CI_GPG_MASTERKEY')]) {
+
+    sh "gpg --import ${CI_GPG_PRIVKEY}"
```
Comment (bakhtin) [inline .jenkinsci/artifacts.groovy]: This one fails on MacOS with the following error: `error sending to agent: Inappropriate ioctl for device`. [There is a proposed solution](https://github.com/keybase/keybase-issues/issues/2798) for this issue, but it didn't help. It then fails with an error: `error sending to agent: No such file or directory`. 
I didn't check the import on Linux though. Make sure it works on Linux too.

Comment (Stayer) [reply .jenkinsci/artifacts.groovy]: I found solution on this page: https://myshittycode.com/2017/08/07/maven-gpg-plugin-prevent-signing-prompt-or-gpg-signing-failed-no-such-file-or-directory-error/

Summary: added "--pinentry-mode loopback" arguments on signing

File: .jenkinsci/artifacts.groovy [hunk 5]
```diff
@@ -11,17 +11,27 @@ def uploadArtifacts(filePaths, uploadPath, artifactServers=['artifact.soramitsu.
   }
   def shaSumBinary = 'sha256sum'
   def md5SumBinary = 'md5sum'
+  def gpgKeyBinary = 'gpg --armor --detach-sign --no-tty --batch --yes --passphrase-fd 0'
   if (agentType == 'Darwin') {
     shaSumBinary = 'shasum -a 256'
     md5SumBinary = 'md5 -r'
+    gpgKeyBinary = gpgKeyBinary = 'GPG_TTY=\$(tty) gpg --pinentry-mode loopback --armor --detach-sign --no-tty --batch --yes --passphrase-fd 0'
```
Comment (bakhtin) [inline .jenkinsci/artifacts.groovy]: Please, fix typo (`gpgKeyBinary = gpgKeyBinary`). Overall, LGTM. Wait for another peer to review before merging.

**Pull request merged**